app/models/sponsor.rb

```diff
@@ -11,14 +12,26 @@ class Sponsor < ActiveRecord::Base
 
   has_one :address
   has_many :chapters, through: :workshops
-  has_many :workshop_sponsors
-  has_many :workshops, through: :workshop_sponsors
+  has_many :contacts
+  has_many :meetings, -> { order(date_and_time: :desc) }, foreign_key: 'venue_id', inverse_of: :venue
   has_many :member_contacts
-  has_many :contacts, through: :member_contacts, class_name: 'Member', foreign_key: 'member_id'
+  has_many :members, through: :member_contacts
+  has_many :sponsorships, -> { includes([:event]).joins(:event).order('events.date_and_time desc') },
```

Comment:
@despo I'm wondering if the 'name' of this assoc could allude that these are event sponsorships. It would make the `sponsorships_count` method in `SponsorPresenter` clearer.

Reply:
Good point. Even though I do think that of the sponsor association names this is the best, it is indeed on clear that it's related to Events. I would like to eventually rename this to event_sponsorships, and apply the same approach to workshop_sponsors and meetings_sponsors, so that we have workshop_sponsorships and meeting_sponsorships.

For now I will rename only the association to event_sponsorships and also add a `has_many :events` on the sponsor model, that I can use to calculate `sponsorship_count` on the presenter.

Reply:
@despo yep, sounds good